api/v1alpha1/httproutefilter_types.go

```diff
@@ -37,6 +37,12 @@ type HTTPRouteFilterSpec struct {
 
 // HTTPURLRewriteFilter define rewrites of HTTP URL components such as path and host
 type HTTPURLRewriteFilter struct {
+	// Hostname is the value to be used to replace the Host header value during
+	// forwarding.
+	//
+	// +optional
+	// +notImplementedHide
+	Hostname *HTTPHostnameModifier `json:"hostname,omitempty"`
```

Comment:
Nit: How about `Host` instead of `Hostname`? 
```suggestion
	Host *HTTPHostModifier `json:"host,omitempty"`
```

Reply:
The GW-API terminology is hostname: https://gateway.envoyproxy.io/docs/tasks/traffic/http-urlrewrite/#rewrite-host-name, so I think this would be more consistent for users.